integration/header_test.go

```diff
@@ -197,4 +198,27 @@ var _ = Describe("Headers", func() {
 			Expect(resp.Header.Get(HeaderKeySignature)).To(BeEmpty())
 		})
 	})
+
+	Context("Header Limits", func() {
+		Context("when a response header size limit is configured", func() {
+			BeforeEach(func() {
+				testApp = NewUnstartedTestApp(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.Header().Set("Too-Large", strings.Repeat("0123456789", 10))
+					w.WriteHeader(200)
+				}))
+				testState.cfg.MaxResponseHeaderBytes = 80
+				testState.StartGorouterOrFail()
+				testApp.Start()
+				testState.register(testApp.Server, testAppRoute)
+			})
+
+			It("fails with 502 when the app exceeds the limit", func() {
+				req := testState.newRequest(fmt.Sprintf("http://%s", testAppRoute))
+				resp, err := testState.client.Do(req)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(http.StatusBadGateway))
+				Expect(resp.Header).To(HaveKey("X-Cf-Routererror"))
```

Comment:
should we check here also for the error message to make sure the right router error is reported?